src/test/java/com/hyperlocal/server/ShopControllerTest.java

```diff
@@ -156,43 +142,46 @@ public void shouldUpsertCatalog() throws Exception {
     List<Object> addList = Arrays.asList(shopID, "Mango", "Lorem ipsum", "#");
     addCommand.put("serviceName", addList.get(1));
     addCommand.put("serviceDescription", addList.get(2));
     addCommand.put("imageURL", addList.get(3));
-    payload.put("add", new HashMap[]{addCommand});
+    payload.put("add", new HashMap[] { addCommand });
     HashMap<String, Object> editCommand = new HashMap<String, Object>();
     List<Object> editList = Arrays.asList("Apples", "Lorem ipsum", "#", 9000000000L);
     editCommand.put("serviceName", editList.get(0));
     editCommand.put("serviceDescription", editList.get(1));
     editCommand.put("imageURL", editList.get(2));
     editCommand.put("serviceID", editList.get(3));
-    payload.put("edit", new HashMap[]{editCommand});
+    payload.put("edit", new HashMap[] { editCommand });
     HashMap<String, Object> deleteCommand = new HashMap<String, Object>();
     Long deleteServiceID = 9000000000L;
     deleteCommand.put("serviceID", deleteServiceID);
-    payload.put("delete", new HashMap[]{deleteCommand});
+    payload.put("delete", new HashMap[] { deleteCommand });
 
     QueryResult emptyQueryResult = new QueryResult(1L, "Success", resultSet);
 
     HashMap<String, Object> expectedMap = new HashMap<String, Object>();
     expectedMap.put("success", true);
-    
-    when(connection.sendQuery("BEGIN"))
-    .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+
+    when(connection.sendQuery("BEGIN")).thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
     when(connection.sendPreparedStatement(INSERT_CATALOG_STATEMENT, addList))
-    .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+        .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
     when(connection.sendPreparedStatement(UPDATE_CATALOG_STATEMENT, editList))
-    .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+        .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
     when(connection.sendPreparedStatement(DELETE_CATALOG_STATEMENT, Arrays.asList(deleteServiceID)))
-    .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
-    when(connection.sendQuery("COMMIT"))
-    .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+        .thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+    when(connection.sendQuery("COMMIT")).thenReturn(CompletableFuture.completedFuture(emptyQueryResult));
+    when(template.publish("projects/speedy-anthem-217710/topics/testTopic", "1000000000000"))
+        .thenReturn(new AsyncResult<>("DONE"));
 
     /* ACT */
-    CompletableFuture<HashMap<String, Object>> actualMapPromise = controller.upsertCatalog(shopID, new Gson().toJson(payload));
+    CompletableFuture<HashMap<String, Object>> actualMapPromise = controller.upsertCatalog(shopID,
+        new Gson().toJson(payload));
+
+    verify(template).publish("projects/speedy-anthem-217710/topics/testTopic", "1000000000000");
```

Comment:
This verification can go under the `ASSERT` section starting at (new) line 186.

Reply:
Done. Thanks a lot!!!